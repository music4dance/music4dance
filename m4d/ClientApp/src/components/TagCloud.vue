<script setup lang="ts">
import { SongFilter } from "@/models/SongFilter";
import { Tag, TagBucket, type TagInfo } from "@/models/Tag";
import { TagHandler } from "@/models/TagHandler";
import { computed, reactive, ref } from "vue";
import { type ColorVariant } from "bootstrap-vue-next";

class TagButton implements TagInfo {
  public static get buttons() {
    return Tag.tagKeys
      .filter((t) => t !== "dance")
      .map((t) => new TagButton(t, Tag.tagInfo.get(t)!));
  }

  public key: string;
  public variant: ColorVariant;
  public iconName: string;
  public description: string;
  public state: boolean;

  constructor(key: string, info: TagInfo) {
    this.key = key;
    this.variant = key as ColorVariant;
    this.iconName = info.iconName;
    this.description = info.description;
    this.state = true;
  }
}

const props = defineProps<{
  tags: Tag[];
  hideFilter?: boolean;
  songFilter?: SongFilter;
  user?: string;
}>();

const modalVisible = ref(false);
const currentTag = ref<TagHandler>(new TagHandler({ tag: Tag.fromString("Placeholder:Other") }));
const tagButtons = reactive(TagButton.buttons);
const filter = ref(6);

// INT-TODO: Test for custom variants
// const topButtonVariant: ColorVariant = "music";
// console.log(topButtonVariant);

const activeTags = computed(() => {
  const tags = tagButtons.filter((b) => b.state).map((b) => b.key);
  return tags;
});

const tagBuckets = computed(() => {
  const active = activeTags;
  const tags = props.tags.filter(
    (t) => active.value.includes(t.category.toLowerCase()) && t.count && t.count > 0,
  );
  let bucketized = TagBucket.bucketize(tags);
  if (filter.value < 10) {
    const threshold = 10 - filter.value;
    bucketized = TagBucket.bucketize(bucketized.filter((b) => b.bucket! > threshold));
  }
  return bucketized.sort((a, b) => a.value.localeCompare(b.value));
});

function classesForTag(tag: TagBucket): string[] {
  return ["cloud-" + tag.bucket];
}

function showModal(key: string): void {
  currentTag.value = getTagHandler(Tag.fromString(key));
  modalVisible.value = true;
}

function getTagHandler(tag: Tag): TagHandler {
  return new TagHandler({ tag, user: props.user, filter: props.songFilter });
}
</script>

<template>
  <div>
    <div v-if="!hideFilter" class="row" style="margin-bottom: 0.5rem">
      <BButtonGroup size="sm" class="col-sm">
        <BButton
          v-for="btn in tagButtons"
          :id="'filter-' + btn.key"
          :key="btn.key"
          v-model:pressed="btn.state"
          :variant="btn.variant"
        >
          <TagIcon :name="btn.iconName" />
          {{ btn.description }}
          <IBiCheckCircle v-if="btn.state" />
          <IBiCircle v-else />
        </BButton>
      </BButtonGroup>
      <div style="width: 250px" class="mx-auto col-sm">
        <BFormInput id="strictness" v-model="filter" type="range" min="2" max="10" step="1" />
        <label
          for="strictness"
          class="d-flex justify-content-between"
          style="margin-bottom: 0; margin-top: -0.75rem"
        >
          <span>less</span><span class="ms-auto">more</span>
        </label>
      </div>
    </div>
    <div v-if="tagBuckets.length > 0">
      <BBadge
        v-for="tag in tagBuckets"
        :key="tag.key"
        :variant="tag.variant"
        style="margin-inline-end: 0.25em; margin-bottom: 0.25em"
        :class="classesForTag(tag)"
        href="#"
        role="button"
        @click="showModal(tag.key)"
      >
        <TagIcon :name="tag.icon!" />{{ tag.value }}
      </BBadge>
    </div>
    <div v-else>
      <h4>Please select one or more tag classes (style, tempo, musical genre, oother)</h4>
    </div>
    <TagModal v-model="modalVisible" :tag-handler="currentTag as TagHandler" />
  </div>
</template>

<style scoped>
.cloud-0 {
  font-size: 80%;
}

.cloud-1 {
  font-size: 90%;
}

.cloud-2 {
  font-size: 100%;
}

.cloud-3 {
  font-size: 110%;
}

.cloud-4 {
  font-size: 120%;
}

.cloud-5 {
  font-size: 130%;
}

.cloud-6 {
  font-size: 140%;
}

.cloud-7 {
  font-size: 150%;
}

.cloud-8 {
  font-size: 160%;
}

.cloud-9 {
  font-size: 170%;
}
</style>
