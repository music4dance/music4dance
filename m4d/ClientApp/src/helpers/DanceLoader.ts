import { DanceDatabase } from "@/models/DanceDatabase";

export function loadDancesFromString(s: string): DanceDatabase {
  const db = DanceDatabase.load(s);
  if (!db) {
    throw new Error("Dance Database not loaded as expected");
  }
  return db;
}
