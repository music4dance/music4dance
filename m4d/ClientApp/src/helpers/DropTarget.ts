import { ServiceMatcher } from "./ServiceMatcher";

const matcher: ServiceMatcher = new ServiceMatcher();

export async function checkServiceAndAdd(input: string, warn?: boolean): Promise<void> {
  const service = matcher.match(input);
  if (!service) {
    return;
  }
  const found = await checkService(input);

  if (!found) {
    let okay = false;
    if (warn) {
      // INT-TODO: Figure out if there is a way to do this with BSV-Next
      //   okay = await this.$bvModal.msgBoxConfirm(
      //     `It looks like you may have tried to search by ${service.name} id for a song not in the music4dance catalog.
      //  Would you like to add the song?`,
      //     {
      //       title: "Music Service Search?",
      //       okTitle: "Add Song",
      //     },
      //   );

      okay = window.confirm(
        `It looks like you may have tried to search by ${service.name} id for a song not in the music4dance catalog. Would you like to add the song?`,
      );
    }
    if (okay) {
      const id = matcher.parseId(input, service);
      window.location.href = `/song/augment?id=${id}`;
    }
  }
}

export async function checkServiceAndWarn(input: string): Promise<void> {
  return checkServiceAndAdd(input, true);
}

async function checkService(input: string): Promise<boolean> {
  const service = matcher.match(input);
  if (service) {
    try {
      const id = matcher.parseId(input, service);
      if (id) {
        const song = await matcher.findSong(input, true);
        if (song) {
          window.location.href = `/song/details/${song.songHistory.id}`;
          return true;
        }
      }
    } catch {
      // swallow any errors
    }
  }
  return false;
}
