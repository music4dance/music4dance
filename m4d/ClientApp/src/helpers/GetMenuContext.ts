import { MenuContext } from "@/models/MenuContext";
import type { AxiosInstance } from "axios";

declare global {
  interface Window {
    menuContext: MenuContext;
  }
}

let menuContextObject: MenuContext | null = null;

export function getMenuContext(): MenuContext {
  if (!menuContextObject) {
    menuContextObject = window.menuContext
      ? new MenuContext(window.menuContext)
      : new MenuContext();
  }
  return menuContextObject;
}

export function getAxiosXsrf(): AxiosInstance {
  return getMenuContext().axiosXsrf;
}
