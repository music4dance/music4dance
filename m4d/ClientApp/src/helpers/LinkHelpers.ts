import type { DanceInstance } from "@/models/DanceDatabase/DanceInstance";
import type { DanceType } from "@/models/DanceDatabase/DanceType";
import type { TempoRange } from "@/models/DanceDatabase/TempoRange";
import { wordsToKebab } from "./StringHelpers";

export function danceLink(dance: DanceInstance): string {
  return wordsToKebab(dance.shortName);
}

export function tempoLink(dance: DanceInstance | DanceType, tempo: TempoRange): string {
  return `/song/advancedsearch?dances=${dance.baseId}&tempomin=${tempo.min}&tempomax=${tempo.max}&sortorder=Dances`;
}

export function defaultTempoLink(dance: DanceInstance | DanceType): string {
  return tempoLink(dance, dance.tempoRange);
}

export function filteredTempoLink(dance: DanceInstance, filter: string): string {
  return tempoLink(dance, dance.filteredTempo([filter])!);
}
