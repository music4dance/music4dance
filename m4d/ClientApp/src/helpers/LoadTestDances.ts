// @ts-ignore
import dancesJson from "@/assets/dances.json";
// @ts-ignore
import groupsJson from "@/assets/dancegroups.json";

export function loadTestDances(): string {
  return JSON.stringify({ dances: dancesJson, groups: groupsJson });
}
