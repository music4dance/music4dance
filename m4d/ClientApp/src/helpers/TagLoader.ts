import { Tag } from "@/models/Tag";
import { TagDatabase } from "@/models/TagDatabase";
import { TypedJSON } from "typedjson";

export function loadTagsFromString(s: string, includeIncremental?: boolean): TagDatabase {
  const tags = TypedJSON.parseAsArray(s, Tag);
  return new TagDatabase(tags, includeIncremental ? loadIncrementalTags() : undefined);
}
function loadIncrementalTags(): Tag[] | undefined {
  const incrementalString = sessionStorage.getItem("incremental-tags");
  return incrementalString ? TypedJSON.parseAsArray(incrementalString, Tag) : undefined;
}
