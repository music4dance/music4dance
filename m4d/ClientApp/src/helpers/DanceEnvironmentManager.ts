import { DanceDatabase } from "@/models/DanceDatabase/DanceDatabase";
import { loadDancesFromString } from "./DanceLoader";

declare global {
  interface Window {
    danceDatabaseJson?: string;
    danceDatabase: DanceDatabase | undefined;
  }
}

export function safeDanceDatabase(): DanceDatabase {
  if (window.danceDatabase) {
    return window.danceDatabase;
  }

  if (!window.danceDatabaseJson) {
    throw new Error("danceDatabaseJson not defined on window");
  }

  const db = loadDancesFromString(window.danceDatabaseJson);
  if (db) {
    window.danceDatabase = db;
    return db;
  }
  throw new Error("Dance Database not loaded as expected");
}
