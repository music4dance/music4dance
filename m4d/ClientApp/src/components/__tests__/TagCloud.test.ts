import { mount } from "@vue/test-utils";
import { describe, expect, test } from "vitest";
import { modalManagerPlugin } from "bootstrap-vue-next";
import { loadTagsFromString } from "@/helpers/TagLoader";
import TagCloud from "../TagCloud.vue";

const tagsJson = [
  {
    key: "1940S:Other",
    count: 215,
  },
  {
    key: "1949:Other",
    count: 2,
  },
  {
    key: "Jazz:Music",
    count: 132,
  },
  {
    key: "Rock:Music",
    count: 7,
  },
  {
    key: "Changing:Tempo",
    count: 2,
  },
  {
    key: "Contemporary:Style",
    count: 323,
  },
];

const tagDatabase = loadTagsFromString(JSON.stringify(tagsJson));

describe("TagCloud.vue", () => {
  test("checks raw tags", () => {
    const tags = tagDatabase.tags;
    expect(tags.length).toBe(6);
    const tag = tagDatabase.getTag("1940s:Other");
    expect(tag).toBeDefined();
    expect(tag!.key).toBe("1940S:Other");
  });

  test("Renders a simple tag cloud", () => {
    const tags = tagDatabase.tags;
    const wrapper = mount(TagCloud, {
      props: { tags },
      global: { plugins: [modalManagerPlugin] },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("Renders a tag cloud without filters", () => {
    const tags = tagDatabase.tags;
    const wrapper = mount(TagCloud, {
      props: { tags, hideFilter: true },
      global: { plugins: [modalManagerPlugin] },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("Check tag cloud filtering", async () => {
    const tags = tagDatabase.tags;
    const wrapper = mount(TagCloud, {
      props: { tags },
      global: { plugins: [modalManagerPlugin] },
    });

    // Rock is filtered out by strictness
    expect(wrapper.findAll(".text-bg-music").length).toBe(1);

    // Set strictness to include all tags
    const strictness = wrapper.find("#strictness");
    await strictness.setValue(10);

    // Now both music tags are shown
    expect(wrapper.findAll(".text-bg-music").length).toBe(2);

    // Filter out music tags
    const music = wrapper.find("#filter-music");
    expect(music).toBeDefined();
    await music.trigger("click");

    // No music tags are shown
    expect(wrapper.findAll(".text-bg-music").length).toBe(0);

    // But both other tags are shown
    expect(wrapper.findAll(".text-bg-other").length).toBe(2);
  });
});
