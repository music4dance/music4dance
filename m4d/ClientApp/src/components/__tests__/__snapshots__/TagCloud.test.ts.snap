// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TagCloud.vue > Renders a simple tag cloud 1`] = `
"<div data-v-dff412d7=\\"\\">
  <div data-v-dff412d7=\\"\\" class=\\"row\\" style=\\"margin-bottom: 0.5rem;\\">
    <div data-v-dff412d7=\\"\\" class=\\"btn-group btn-group-sm col-sm\\" role=\\"group\\" aria-label=\\"Group\\"><button data-v-dff412d7=\\"\\" class=\\"btn btn-md btn-style active\\" aria-pressed=\\"true\\" autocomplete=\\"off\\" type=\\"button\\" id=\\"filter-style\\"><svg data-v-dff412d7=\\"\\" viewBox=\\"0 0 16 16\\" width=\\"1.2em\\" height=\\"1.2em\\">
          <path fill=\\"currentColor\\" d=\\"M6.5 1A1.5 1.5 0 0 0 5 2.5V3H1.5A1.5 1.5 0 0 0 0 4.5v8A1.5 1.5 0 0 0 1.5 14h13a1.5 1.5 0 0 0 1.5-1.5v-8A1.5 1.5 0 0 0 14.5 3H11v-.5A1.5 1.5 0 0 0 9.5 1zm0 1h3a.5.5 0 0 1 .5.5V3H6v-.5a.5.5 0 0 1 .5-.5m1.886 6.914L15 7.151V12.5a.5.5 0 0 1-.5.5h-13a.5.5 0 0 1-.5-.5V7.15l6.614 1.764a1.5 1.5 0 0 0 .772 0M1.5 4h13a.5.5 0 0 1 .5.5v1.616L8.129 7.948a.5.5 0 0 1-.258 0L1 6.116V4.5a.5.5 0 0 1 .5-.5\\"></path>
        </svg> style <svg data-v-dff412d7=\\"\\" viewBox=\\"0 0 16 16\\" width=\\"1.2em\\" height=\\"1.2em\\">
          <g fill=\\"currentColor\\">
            <path d=\\"M8 15A7 7 0 1 1 8 1a7 7 0 0 1 0 14m0 1A8 8 0 1 0 8 0a8 8 0 0 0 0 16\\"></path>
            <path d=\\"M10.97 4.97a.235.235 0 0 0-.02.022L7.477 9.417L5.384 7.323a.75.75 0 0 0-1.06 1.06L6.97 11.03a.75.75 0 0 0 1.079-.02l3.992-4.99a.75.75 0 0 0-1.071-1.05\\"></path>
          </g>
        </svg></button><button data-v-dff412d7=\\"\\" class=\\"btn btn-md btn-tempo active\\" aria-pressed=\\"true\\" autocomplete=\\"off\\" type=\\"button\\" id=\\"filter-tempo\\"><svg data-v-dff412d7=\\"\\" viewBox=\\"0 0 16 16\\" width=\\"1.2em\\" height=\\"1.2em\\">
          <g fill=\\"currentColor\\">
            <path d=\\"M8 3.5a.5.5 0 0 0-1 0V9a.5.5 0 0 0 .252.434l3.5 2a.5.5 0 0 0 .496-.868L8 8.71z\\"></path>
            <path d=\\"M8 16A8 8 0 1 0 8 0a8 8 0 0 0 0 16m7-8A7 7 0 1 1 1 8a7 7 0 0 1 14 0\\"></path>
          </g>
        </svg> tempo <svg data-v-dff412d7=\\"\\" viewBox=\\"0 0 16 16\\" width=\\"1.2em\\" height=\\"1.2em\\">
          <g fill=\\"currentColor\\">
            <path d=\\"M8 15A7 7 0 1 1 8 1a7 7 0 0 1 0 14m0 1A8 8 0 1 0 8 0a8 8 0 0 0 0 16\\"></path>
            <path d=\\"M10.97 4.97a.235.235 0 0 0-.02.022L7.477 9.417L5.384 7.323a.75.75 0 0 0-1.06 1.06L6.97 11.03a.75.75 0 0 0 1.079-.02l3.992-4.99a.75.75 0 0 0-1.071-1.05\\"></path>
          </g>
        </svg></button><button data-v-dff412d7=\\"\\" class=\\"btn btn-md btn-music active\\" aria-pressed=\\"true\\" autocomplete=\\"off\\" type=\\"button\\" id=\\"filter-music\\"><svg data-v-dff412d7=\\"\\" viewBox=\\"0 0 16 16\\" width=\\"1.2em\\" height=\\"1.2em\\">
          <g fill=\\"currentColor\\">
            <path d=\\"M12 13c0 1.105-1.12 2-2.5 2S7 14.105 7 13s1.12-2 2.5-2s2.5.895 2.5 2\\"></path>
            <path fill-rule=\\"evenodd\\" d=\\"M12 3v10h-1V3z\\"></path>
            <path d=\\"M11 2.82a1 1 0 0 1 .804-.98l3-.6A1 1 0 0 1 16 2.22V4l-5 1z\\"></path>
            <path fill-rule=\\"evenodd\\" d=\\"M0 11.5a.5.5 0 0 1 .5-.5H4a.5.5 0 0 1 0 1H.5a.5.5 0 0 1-.5-.5m0-4A.5.5 0 0 1 .5 7H8a.5.5 0 0 1 0 1H.5a.5.5 0 0 1-.5-.5m0-4A.5.5 0 0 1 .5 3H8a.5.5 0 0 1 0 1H.5a.5.5 0 0 1-.5-.5\\"></path>
          </g>
        </svg> musical genre <svg data-v-dff412d7=\\"\\" viewBox=\\"0 0 16 16\\" width=\\"1.2em\\" height=\\"1.2em\\">
          <g fill=\\"currentColor\\">
            <path d=\\"M8 15A7 7 0 1 1 8 1a7 7 0 0 1 0 14m0 1A8 8 0 1 0 8 0a8 8 0 0 0 0 16\\"></path>
            <path d=\\"M10.97 4.97a.235.235 0 0 0-.02.022L7.477 9.417L5.384 7.323a.75.75 0 0 0-1.06 1.06L6.97 11.03a.75.75 0 0 0 1.079-.02l3.992-4.99a.75.75 0 0 0-1.071-1.05\\"></path>
          </g>
        </svg></button><button data-v-dff412d7=\\"\\" class=\\"btn btn-md btn-other active\\" aria-pressed=\\"true\\" autocomplete=\\"off\\" type=\\"button\\" id=\\"filter-other\\"><svg data-v-dff412d7=\\"\\" viewBox=\\"0 0 16 16\\" width=\\"1.2em\\" height=\\"1.2em\\">
          <g fill=\\"currentColor\\">
            <path d=\\"M6 4.5a1.5 1.5 0 1 1-3 0a1.5 1.5 0 0 1 3 0m-1 0a.5.5 0 1 0-1 0a.5.5 0 0 0 1 0\\"></path>
            <path d=\\"M2 1h4.586a1 1 0 0 1 .707.293l7 7a1 1 0 0 1 0 1.414l-4.586 4.586a1 1 0 0 1-1.414 0l-7-7A1 1 0 0 1 1 6.586V2a1 1 0 0 1 1-1m0 5.586l7 7L13.586 9l-7-7H2z\\"></path>
          </g>
        </svg> other <svg data-v-dff412d7=\\"\\" viewBox=\\"0 0 16 16\\" width=\\"1.2em\\" height=\\"1.2em\\">
          <g fill=\\"currentColor\\">
            <path d=\\"M8 15A7 7 0 1 1 8 1a7 7 0 0 1 0 14m0 1A8 8 0 1 0 8 0a8 8 0 0 0 0 16\\"></path>
            <path d=\\"M10.97 4.97a.235.235 0 0 0-.02.022L7.477 9.417L5.384 7.323a.75.75 0 0 0-1.06 1.06L6.97 11.03a.75.75 0 0 0 1.079-.02l3.992-4.99a.75.75 0 0 0-1.071-1.05\\"></path>
          </g>
        </svg></button></div>
    <div data-v-dff412d7=\\"\\" style=\\"width: 250px;\\" class=\\"mx-auto col-sm\\"><input data-v-dff412d7=\\"\\" id=\\"strictness\\" class=\\"form-range\\" type=\\"range\\" min=\\"2\\" max=\\"10\\" step=\\"1\\"><label data-v-dff412d7=\\"\\" for=\\"strictness\\" class=\\"d-flex justify-content-between\\" style=\\"margin-bottom: 0px; margin-top: -0.75rem;\\"><span data-v-dff412d7=\\"\\">less</span><span data-v-dff412d7=\\"\\" class=\\"ml-auto\\">more</span></label></div>
  </div>
  <div data-v-dff412d7=\\"\\"><a data-v-dff412d7=\\"\\" class=\\"badge text-bg-other text-decoration-none cloud-3\\" href=\\"#\\" target=\\"_self\\" style=\\"margin-inline-end: 0.25em; margin-bottom: 0.25em;\\" role=\\"button\\"><svg data-v-dff412d7=\\"\\" viewBox=\\"0 0 16 16\\" width=\\"1.2em\\" height=\\"1.2em\\">
        <g fill=\\"currentColor\\">
          <path d=\\"M6 4.5a1.5 1.5 0 1 1-3 0a1.5 1.5 0 0 1 3 0m-1 0a.5.5 0 1 0-1 0a.5.5 0 0 0 1 0\\"></path>
          <path d=\\"M2 1h4.586a1 1 0 0 1 .707.293l7 7a1 1 0 0 1 0 1.414l-4.586 4.586a1 1 0 0 1-1.414 0l-7-7A1 1 0 0 1 1 6.586V2a1 1 0 0 1 1-1m0 5.586l7 7L13.586 9l-7-7H2z\\"></path>
        </g>
      </svg>1940S</a><a data-v-dff412d7=\\"\\" class=\\"badge text-bg-style text-decoration-none cloud-6\\" href=\\"#\\" target=\\"_self\\" style=\\"margin-inline-end: 0.25em; margin-bottom: 0.25em;\\" role=\\"button\\"><svg data-v-dff412d7=\\"\\" viewBox=\\"0 0 16 16\\" width=\\"1.2em\\" height=\\"1.2em\\">
        <path fill=\\"currentColor\\" d=\\"M6.5 1A1.5 1.5 0 0 0 5 2.5V3H1.5A1.5 1.5 0 0 0 0 4.5v8A1.5 1.5 0 0 0 1.5 14h13a1.5 1.5 0 0 0 1.5-1.5v-8A1.5 1.5 0 0 0 14.5 3H11v-.5A1.5 1.5 0 0 0 9.5 1zm0 1h3a.5.5 0 0 1 .5.5V3H6v-.5a.5.5 0 0 1 .5-.5m1.886 6.914L15 7.151V12.5a.5.5 0 0 1-.5.5h-13a.5.5 0 0 1-.5-.5V7.15l6.614 1.764a1.5 1.5 0 0 0 .772 0M1.5 4h13a.5.5 0 0 1 .5.5v1.616L8.129 7.948a.5.5 0 0 1-.258 0L1 6.116V4.5a.5.5 0 0 1 .5-.5\\"></path>
      </svg>Contemporary</a><a data-v-dff412d7=\\"\\" class=\\"badge text-bg-music text-decoration-none cloud-0\\" href=\\"#\\" target=\\"_self\\" style=\\"margin-inline-end: 0.25em; margin-bottom: 0.25em;\\" role=\\"button\\"><svg data-v-dff412d7=\\"\\" viewBox=\\"0 0 16 16\\" width=\\"1.2em\\" height=\\"1.2em\\">
        <g fill=\\"currentColor\\">
          <path d=\\"M12 13c0 1.105-1.12 2-2.5 2S7 14.105 7 13s1.12-2 2.5-2s2.5.895 2.5 2\\"></path>
          <path fill-rule=\\"evenodd\\" d=\\"M12 3v10h-1V3z\\"></path>
          <path d=\\"M11 2.82a1 1 0 0 1 .804-.98l3-.6A1 1 0 0 1 16 2.22V4l-5 1z\\"></path>
          <path fill-rule=\\"evenodd\\" d=\\"M0 11.5a.5.5 0 0 1 .5-.5H4a.5.5 0 0 1 0 1H.5a.5.5 0 0 1-.5-.5m0-4A.5.5 0 0 1 .5 7H8a.5.5 0 0 1 0 1H.5a.5.5 0 0 1-.5-.5m0-4A.5.5 0 0 1 .5 3H8a.5.5 0 0 1 0 1H.5a.5.5 0 0 1-.5-.5\\"></path>
        </g>
      </svg>Jazz</a></div>
  <!--teleport start-->
  <!--teleport end-->
</div>"
`;

exports[`TagCloud.vue > Renders a tag cloud without filters 1`] = `
"<div data-v-dff412d7=\\"\\">
  <!--v-if-->
  <div data-v-dff412d7=\\"\\"><a data-v-dff412d7=\\"\\" class=\\"badge text-bg-other text-decoration-none cloud-3\\" href=\\"#\\" target=\\"_self\\" style=\\"margin-inline-end: 0.25em; margin-bottom: 0.25em;\\" role=\\"button\\"><svg data-v-dff412d7=\\"\\" viewBox=\\"0 0 16 16\\" width=\\"1.2em\\" height=\\"1.2em\\">
        <g fill=\\"currentColor\\">
          <path d=\\"M6 4.5a1.5 1.5 0 1 1-3 0a1.5 1.5 0 0 1 3 0m-1 0a.5.5 0 1 0-1 0a.5.5 0 0 0 1 0\\"></path>
          <path d=\\"M2 1h4.586a1 1 0 0 1 .707.293l7 7a1 1 0 0 1 0 1.414l-4.586 4.586a1 1 0 0 1-1.414 0l-7-7A1 1 0 0 1 1 6.586V2a1 1 0 0 1 1-1m0 5.586l7 7L13.586 9l-7-7H2z\\"></path>
        </g>
      </svg>1940S</a><a data-v-dff412d7=\\"\\" class=\\"badge text-bg-style text-decoration-none cloud-6\\" href=\\"#\\" target=\\"_self\\" style=\\"margin-inline-end: 0.25em; margin-bottom: 0.25em;\\" role=\\"button\\"><svg data-v-dff412d7=\\"\\" viewBox=\\"0 0 16 16\\" width=\\"1.2em\\" height=\\"1.2em\\">
        <path fill=\\"currentColor\\" d=\\"M6.5 1A1.5 1.5 0 0 0 5 2.5V3H1.5A1.5 1.5 0 0 0 0 4.5v8A1.5 1.5 0 0 0 1.5 14h13a1.5 1.5 0 0 0 1.5-1.5v-8A1.5 1.5 0 0 0 14.5 3H11v-.5A1.5 1.5 0 0 0 9.5 1zm0 1h3a.5.5 0 0 1 .5.5V3H6v-.5a.5.5 0 0 1 .5-.5m1.886 6.914L15 7.151V12.5a.5.5 0 0 1-.5.5h-13a.5.5 0 0 1-.5-.5V7.15l6.614 1.764a1.5 1.5 0 0 0 .772 0M1.5 4h13a.5.5 0 0 1 .5.5v1.616L8.129 7.948a.5.5 0 0 1-.258 0L1 6.116V4.5a.5.5 0 0 1 .5-.5\\"></path>
      </svg>Contemporary</a><a data-v-dff412d7=\\"\\" class=\\"badge text-bg-music text-decoration-none cloud-0\\" href=\\"#\\" target=\\"_self\\" style=\\"margin-inline-end: 0.25em; margin-bottom: 0.25em;\\" role=\\"button\\"><svg data-v-dff412d7=\\"\\" viewBox=\\"0 0 16 16\\" width=\\"1.2em\\" height=\\"1.2em\\">
        <g fill=\\"currentColor\\">
          <path d=\\"M12 13c0 1.105-1.12 2-2.5 2S7 14.105 7 13s1.12-2 2.5-2s2.5.895 2.5 2\\"></path>
          <path fill-rule=\\"evenodd\\" d=\\"M12 3v10h-1V3z\\"></path>
          <path d=\\"M11 2.82a1 1 0 0 1 .804-.98l3-.6A1 1 0 0 1 16 2.22V4l-5 1z\\"></path>
          <path fill-rule=\\"evenodd\\" d=\\"M0 11.5a.5.5 0 0 1 .5-.5H4a.5.5 0 0 1 0 1H.5a.5.5 0 0 1-.5-.5m0-4A.5.5 0 0 1 .5 7H8a.5.5 0 0 1 0 1H.5a.5.5 0 0 1-.5-.5m0-4A.5.5 0 0 1 .5 3H8a.5.5 0 0 1 0 1H.5a.5.5 0 0 1-.5-.5\\"></path>
        </g>
      </svg>Jazz</a></div>
  <!--teleport start-->
  <!--teleport end-->
</div>"
`;
